### PR TITLE
fix(geocoder): HERE API bugs

### PR DESCRIPTION
**Pull request by daniel-heppner-ibigroup**
Fixes some issues with the HERE API.
- Certain API responses could cause a crash.
- Boundary is now calculated internally when a focusPoint is also provided.

### CODE REVIEW

File: packages/geocoder/src/apis/here/index.ts [hunk 1]
```diff
@@ -66,6 +72,22 @@ function autocomplete({
...
+    const res = await run({
+      options,
+      query,
+      url: AUTOCOMPLETE_URL
+    });
+    if (boundary?.rect) {
+      // HERE does not support a boundary when you use a focus point
+      // This workaround filters the results internally to the boundary
+      res.items = res.items.filter(checkItemInBoundary(boundary))
+      return res
```
Comment (miles-grant-ibigroup) [inline packages/geocoder/src/apis/here/index.ts:87]: We can probably remove this line right?

File: packages/geocoder/src/geocoders/here.ts [hunk 2]
```diff
@@ -73,8 +73,9 @@ export default class HereGeocoder extends Geocoder {
     const { items } = response;
     return {
       features: items
-      // Here can return continued query suggestions, which we do not support.
-      ?.filter(item => item.resultType !== "chainQuery")
+      // Here has various types of responses, some of which are not locations.
+      // We only want the actual places, so throw out any without a position.
+      ?.filter(item => item.position !== undefined)
```
Comment (miles-grant-ibigroup) [inline packages/geocoder/src/geocoders/here.ts:78]:
```suggestion
      ?.filter(item => !!item.position)
```
Does this also work?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Looks really good! Just a few small clean up things

File: packages/geocoder/src/geocoders/here.ts (hunk 2 above)
Comment (daniel-heppner-ibigroup) [reply packages/geocoder/src/geocoders/here.ts:78]: Sure, I was being unnecessarily verbose.

**Review by miles-grant-ibigroup — APPROVED** [review]
Looking good!

**Review by binh-dam-ibigroup — APPROVED** [review]

**Pull request merged**